### PR TITLE
Application Broker with one service

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- Application Broker has only one k8s Service
- all namespaced ServiceBroker instances are configured to the same Service with basepath as namespace

Related: #3814

### CODE REVIEW

File: components/application-broker/README.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ Before each commit, use the `before-commit.sh` script or the `make build` comman
 |-----|---------|--------|------------|
 |**APP_PORT** | NO | `8080` | The port on which the HTTP server listens |
 |**APP_BROKER_RELIST_DURATION_WINDOW** | YES | - | Time period after which the AB synchronizes with the Service Catalog if a new Application is added. In case more than one Application is added, synchronization is performed only once. |
+| **APP_SERVICE_NAME** | YES | - | The name of the Kubernetes Service, which exposes the broker API |
```
Comment (polskikiel) [inline components/application-broker/README.md]:
```suggestion
| **APP_SERVICE_NAME** | YES | - | The name of the Kubernetes Service, which exposes the Service Brokers API |
```

File: components/application-broker/internal/nsbroker/facade.go [hunk 2]
```diff
@@ -43,97 +41,55 @@ type brokerSyncer interface {
...
+		abSelectorValue:  abSelectorValue,
+		abTargetPort:     abTargetPort,
+		workingNamespace: workingNamespace,
+		brokerSyncer:     brokerSyncer,
+		serviceName:      serviceName,
+		log:              log.WithField("service", "nsbroker-facade"),
 	}
 }
 
 // Create creates k8s service and ServiceBroker. Errors don't stop execution of method. AlreadyExist errors are ignored.
```
Comment (polskikiel) [inline components/application-broker/internal/nsbroker/facade.go]:
```suggestion
// Create creates ServiceBroker. Errors don't stop execution of method. AlreadyExist errors are ignored.
```

File: components/application-broker/internal/nsbroker/facade.go [hunk 3]
```diff
@@ -171,22 +128,15 @@ func (f *Facade) createServiceBroker(svcURL, namespace string) (*v1beta1.Service
...
 // Delete removes ServiceBroker and Facade. Errors don't stop execution of method. NotFound errors are ignored.
 func (f *Facade) Delete(destinationNs string) error {
-	var resultErr *multierror.Error
-	if err := f.brokerGetter.ServiceBrokers(destinationNs).Delete(NamespacedBrokerName, nil); err != nil {
+	err := f.brokerGetter.ServiceBrokers(destinationNs).Delete(NamespacedBrokerName, nil)
+	switch {
+	case k8serrors.IsNotFound(err):
+		return nil
+	case err != nil:
 		f.log.Warnf("Deletion of namespaced-broker for namespace [%s] results in error: [%s]. NotFound errors will be ignored. ", destinationNs, err)
```
Comment (polskikiel) [inline components/application-broker/internal/nsbroker/facade.go]:
```suggestion
		f.log.Warnf("Deletion of namespaced-broker for namespace [%s] results in error: [%s].", destinationNs, err)
```

File: components/application-broker/internal/nsbroker/facade.go [hunk 4]
```diff
@@ -171,22 +128,15 @@ func (f *Facade) createServiceBroker(svcURL, namespace string) (*v1beta1.Service
...
+	return err
 
-	resultErr = f.filterOutMultiError(resultErr, f.ignoreIsNotFound)
-	if resultErr == nil {
-		return nil
-	}
-	return resultErr
 }
 
 // Exist check if ServiceBroker and Service exist.
```
Comment (polskikiel) [inline components/application-broker/internal/nsbroker/facade.go]:
```suggestion
// Exist check if ServiceBroker exists.
```

File: components/application-broker/internal/nsbroker/facade_legacy_test.go [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+)
+
+//go:generate mockery -name=serviceNameProvider -output=automock -outpkg=automock -case=underscore
+type serviceNameProvider interface {
+	GetServiceNameForNsBroker(ns string) string
+}
+
+// legacyFacade is responsible for creation k8s objects for namespaced broker
+// like it was done in production code before a switch to one k8s service for the application broker.
+// The implementation is a copy of old Facade with removed unnecessary code (broker sync, logging)
```
Comment (polskikiel) [inline components/application-broker/internal/nsbroker/facade_legacy_test.go]: Please add a comment that it's used only in tests

File: components/application-broker/internal/nsbroker/migration.go [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+	if broker.Spec.URL == expectedURL {
+		return nil
+	}
+
+	existingServiceName, err := s.getServiceNameFromBrokerURL(broker.Spec.URL)
+	if err != nil {
+		return errors.Wrapf(err, "while getting service name from URL in namespace %s", broker.Namespace)
+	}
+
+	if broker.Spec.URL == expectedURL {
```
Comment (polskikiel) [inline components/application-broker/internal/nsbroker/migration.go]: please remove doubled check

Comment (piotrmiskiewicz) [reply components/application-broker/internal/nsbroker/migration.go]: removed

File: components/application-broker/internal/nsbroker/facade_legacy_test.go (hunk 5 above)
Comment (piotrmiskiewicz) [reply components/application-broker/internal/nsbroker/facade_legacy_test.go]: added comment

File: components/application-broker/README.md [hunk 7]
```diff
@@ -30,6 +30,7 @@ Before each commit, use the `before-commit.sh` script or the `make build` comman
 |-----|---------|--------|------------|
 |**APP_PORT** | NO | `8080` | The port on which the HTTP server listens |
 |**APP_BROKER_RELIST_DURATION_WINDOW** | YES | - | Time period after which the AB synchronizes with the Service Catalog if a new Application is added. In case more than one Application is added, synchronization is performed only once. |
+| **APP_SERVICE_NAME** | YES | - | The name of the Kubernetes Service, which exposes the Service Brokers API |
```
Comment (klaudiagrz) [inline components/application-broker/README.md]:
```suggestion
| **APP_SERVICE_NAME** | YES | - | The name of the Kubernetes service which exposes the Service Brokers API |
```

**Pull request merged**